### PR TITLE
[ntuple] Rename NTuple "root field" to "field zero"

### PR DESCRIPTION
**Pull request by mxxo**
See naming discussion here: https://github.com/root-project/root/pull/5768#discussion_r434156785
Also implemented convenience `GetFieldZeroId` function.

### CODE REVIEW

File: tree/ntuple/v7/inc/ROOT/RField.hxx [hunk 1]
```diff
@@ -248,9 +248,9 @@ public:
 
 
 /// The container field for an ntuple model, which itself has no physical representation
-class RFieldRoot : public Detail::RFieldBase {
+class RFieldTopLevel : public Detail::RFieldBase {
```
Comment (pcanal) [inline tree/ntuple/v7/inc/ROOT/RField.hxx:251]: It feels awkward to use both "FieldTopLevel" and "TopLevelField" ... unless there is a good reason to.   What am I missing?

Comment (mxxo) [reply tree/ntuple/v7/inc/ROOT/RField.hxx:251]: No concrete reason, this was consistent with other RFieldBase implementors (e.g. `RFieldVector` `RFieldArray`). I can modify those names too if you'd like.

Comment (jblomer) [reply tree/ntuple/v7/inc/ROOT/RField.hxx:251]: I think in this context it is consistent.  We can discuss renaming all the field classes (in a different PR) to `RVectorField`, `RArrayField`, etc.

Comment (mxxo) [reply tree/ntuple/v7/inc/ROOT/RField.hxx:251]: > renaming all the field classes (in a different PR)

Just opened #5831 for this set of changes.

Comment (mxxo) [reply tree/ntuple/v7/inc/ROOT/RField.hxx]: I changed things to use `RHeadField` everywhere in a0b6055 
(following the discussion here: https://github.com/root-project/root/pull/5768#discussion_r436809933)

**Comment by jblomer** [comment]
Many thanks, looks good to me.  Please rebase and we'll merge it.

**Comment by mxxo** [comment]
Based on our meeting, I'll rename this to `FieldZero` and push.

**Pull request merged**